riff-raff/app/persistence/TargetDynamoRepository.scala

```diff
@@ -0,0 +1,35 @@
+package persistence
+
+import ci.Target
+import com.amazonaws.services.dynamodbv2.model.PutItemResult
+import com.gu.scanamo.Table
+import org.joda.time.DateTime
+
+case class TargetId(id: String, targetKey: String, region: String, stack: String, app: String, projectName: String, lastSeen: DateTime)
+object TargetId {
+  def apply(tgt: Target, projectName: String, lastSeen: DateTime): TargetId =
+    TargetId(id(tgt, projectName), targetKey(tgt), tgt.region, tgt.stack, tgt.app, projectName, lastSeen)
+  /* Concatenating to make a composite key is pretty 💩, so using 💩 as a separator. #UTF8FTW */
+  def targetKey(tgt: Target) = Seq(tgt.region,tgt.stack,tgt.app).mkString("💩")
+  def id(tgt: Target, projectName: String) = Seq(tgt.region,tgt.stack,tgt.app,projectName).mkString("💩")
```

Comment:
Why does the `projectName` need to be part of the key? 

It at least feels separate enough that it could be a range key, rather than being part of the hash key, so that you could not have the separate `targetKey` field.

Reply:
It has the benefit that if the same RSSA is seen again then it will effectively be upserted and the `lastSeen` will be updated due to the uniqueness constraint.

Happy to have that Dynamo conversation if you're around.